lib/net/http.rb

```diff
@@ -1180,7 +1180,8 @@ def proxy_port
     # The username of the proxy server, if one is configured.
     def proxy_user
       if ENVIRONMENT_VARIABLE_IS_MULTIUSER_SAFE && @proxy_from_env
-        proxy_uri&.user
+        user = proxy_uri&.user
+        CGI.unescape(user) unless user.nil?
```

Comment:
As cgi.rb isn't required here, it results in `uninitialized constant Net::HTTP::CGI` error.

Reply:
And why `unless user.nil?` instead of `if user`?

Reply:
@nobu Thanks for the review, changed it to `if user` and `if pass` respectively and added a `require 'cgi'` at the top.

Reply:
Maybe `require "cgi/escape"` is enough?

Reply:
@mame / @nobu  Changed it to `require "cgi/escape"`!

Reply:
Wouldn't it be `require "cgi/util"` though? And what is the difference between `CGI::unescape` and `CGI.unescape`

Reply:
Changed it to `require "cgi/util"`. Hope that is alright.

Reply:
As for method calls, `::` and `.` are equivalent.
`.` seems preferred in these days though.

And, as `CGI` is used only when proxy env is given, it can be deferred to that cases only.

Reply:
Thank you @nobu. I don't understand the following aspect completely:

> And, as CGI is used only when proxy env is given, it can be deferred to that cases only.

I don't know how the Ruby dependency loading works exactly (I was wondering that the tests didn't catch `uninitialized constant Net::HTTP::CGI`) you described in the first place.

Should I use `autoload` for the CGI module (`autoload :CGI, 'cgi/util'`), or should I potentially require it someplace else, e.g.:

- https://github.com/ruby/net-http/blob/master/lib/net/http.rb#L650-L655

- https://github.com/ruby/net-http/blob/master/lib/net/http.rb#L1177-L1182

Reply:
@nobu Moved back to the `.` syntax and moved the require to the conditionals.